### ISSUE TITLE
Unexpected visible focus for mouse users

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#8366]
In general, mouse users should not see the visible focus indicator that is added for keyboard-only users. We've introduced this feature with #6879, but there are a few occassions where mouse users still see the visual focus: 
- in case a OS command is used (i.e. CMD+TAB to navigate to another application)
- in case the user enters a form input

With the new b2b my account, where multiple forms are used, this becomes more and more visible, hence we like to get this out of he way soon.

**Comment by tobi-or-not-tobi** [comment]
QA Steps:

First: 
- use mouse to navigate to a form field (i.e. search box)
- ensure that the visual focus (blue outline) does not appear while typing

Second:
- navigate the storefront by mouse
- use a OS or browser shortcut (i.e. CMS+TAB or CMD-F)
- ensure that the visual focus does not appear

While testing the above, and in between use the keyboard to navigate (typically tab or shift-tab), the visual focus should be visible.
